### PR TITLE
[JUJU-193] [2.9] Populate supported assumes features for k8s provider

### PR DESCRIPTION
**Pull request by achilleasa**
This PR implements the `environs.SupportedFeaturesEnumerator` interface for the k8s provider which will allow us to write charms that request a particular `k8s-api` version via an `assumes` expression.

As with the previous PRs in this stream of work (support for assumes), these changes are also behind the `charm-assumes` controller feature flag.

## IAAS test
 
```sh
$ juju bootstrap lxd test --no-gui
$ juju show-model | grep -A42 'features:'

$  juju controller-config 'features=[charm-assumes]'
$ juju show-model | grep -A42 'features:'
  supported-features:
  - name: juju
    description: the version of Juju used by the model
    version: 2.9.19.1
```

## CAAS test
 
```sh
# If testing on microk8s
$ make microk8s-operator-update

$ juju bootstrap microk8s mk8s --no-gui
$ juju add-model foo
$ juju show-model | grep -A42 'features:'

$  juju controller-config 'features=[charm-assumes]'
$ juju show-model | grep -A42 'features:'
  supported-features:
  - name: juju
    description: the version of Juju used by the model
    version: 2.9.19
  - name: k8s-api
    description: the Kubernetes API lets charms query and manipulate the state of
      API objects in a Kubernetes cluster
    version: 1.21.0
```

## Documentation changes

We need to create and populate the contents of: https://juju.is/docs/olm/supported-features

### CODE REVIEW

File: core/assumes/sat_checker.go [hunk 1]
```diff
@@ -14,7 +14,10 @@ var (
 	// generate better error messages when an "assumes" expression requests a
 	// feature that is not included in the feature set supported by the
 	// current model.
-	userFriendlyFeatureDescriptions = map[string]string{}
+	UserFriendlyFeatureDescriptions = map[string]string{
```
Comment (juanmanuel-tirado) [inline core/assumes/sat_checker.go:17]: Maybe we need an entity here. In the future the amount of information for the assumes could be more complex.

**Review by juanmanuel-tirado — APPROVED** [review]
LG2M
Just a minor comment on the structure used to define the features.

File: core/assumes/sat_checker.go (hunk 1 above)
Comment (achilleasa) [reply core/assumes/sat_checker.go:17]: This should be good enough for our current use-cases. We can always change this at a later version if we want to display structured information etc.

**Comment by achilleasa** [comment]
$$merge$$

**Pull request merged**